MobiFlight/Joysticks/Joystick.cs

```diff
@@ -96,54 +96,22 @@ protected virtual void EnumerateDevices()
             {
                 this.DIJoystick.GetObjectInfoById(device.ObjectId);
 
-                int offset = device.Offset;
-                int usage = device.Usage;
-                ObjectAspect aspect = device.Aspect;
-                String name = device.Name;
-
                 bool IsAxis = (device.ObjectId.Flags & DeviceObjectTypeFlags.AbsoluteAxis) > 0;
                 bool IsButton = (device.ObjectId.Flags & DeviceObjectTypeFlags.Button) > 0;
                 bool IsPOV = (device.ObjectId.Flags & DeviceObjectTypeFlags.PointOfViewController) > 0;
 
 
                 if (IsAxis && Axes.Count < DIJoystick.Capabilities.AxeCount)
                 {
-                    String axisName;
-                    string axisLabel = "Unknown";
-                    try
-                    {
-                        var OffsetAxisName = GetAxisNameForUsage(usage);
-                        axisName = $"{AxisPrefix} {OffsetAxisName}";
-                        axisLabel = MapDeviceNameToLabel($"{AxisPrefix} {OffsetAxisName}");
-
-                    } catch (ArgumentOutOfRangeException)
-                    {
-                        Log.Instance.log($"Axis can't be mapped: {DIJoystick.Information.InstanceName} Aspect: {aspect} Offset: {offset} Usage: {usage} Axis: {name} Label: {axisLabel}.", LogSeverity.Error);
-                        continue;
-                    }
-                    Axes.Add(new JoystickDevice() { Name = axisName, Label = axisLabel, Type = DeviceType.AnalogInput, JoystickDeviceType = JoystickDeviceType.Axis});
-                    Log.Instance.log($"Added {DIJoystick.Information.InstanceName} Aspect {aspect} + Offset: {offset} Usage: {usage} Axis: {name} Label: {axisLabel}.", LogSeverity.Debug);
-
+                    RegisterAxis(device);
```

Comment:
This method has three sub-methods abstracted for more specific overrides in derived classes and to keep function length down - derived classes do not need to copy the full size of the method if all they want to change is a line or two.